src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -566,7 +569,8 @@ internal void ParseSolution()
             }
 
             // Cache the unique name of each project, and check that we don't have any duplicates.
-            var projectsByUniqueName = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var projectsByUniqueName = new Dictionary<string, ProjectInSolution>(StringComparer.OrdinalIgnoreCase);
+            var projectsByOriginalName = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
I know you followed the previous behavior, but should it be valid to have two projects differing only in casing on a case-sensitive OS?

Reply:
@Forgind I think we need to check the project name independent of the letter case (case-insensitive).

Reply:
Why is that? Just because that's what we've done in the past or is there a design reason behind it?

Reply:
Good question! I am assuming the way it was.
Thinking about this now, if we build the same project/solution in differents OS with different case-sensitive configurations, it would generate differents behaviors, doesn't it?

Reply:
Yes, and it's possible that was the initial reason, but there are places at this point where we distinguish between the two, presumably because users of each OS are confused when something they expect should work doesn't. (Why does it think these two files are the same when they clearly aren't? vs. Why isn't it recognizing my file? I spelled it correctly...) Someone who builds (not just runs) the same code from two OSs should know the relevant differences.

I don't think it particularly matters, since that's an edge case, no one has complained about it (to my knowledge), and that's the way it was previously. Just curious if there was an underlying reason I wasn't familiar with.

Reply:
A good point of view!